plugin/evm/limitorders/limit_order_tx_processor.go

```diff
@@ -79,24 +206,92 @@ func (lotp *limitOrderTxProcessor) HandleOrderBookTx(tx *types.Transaction, bloc
 				BaseAssetQuantity:       baseAssetQuantity,
 				FilledBaseAssetQuantity: 0,
 				Price:                   price,
-				Salt:                    order.Salt.Int64(),
-				Status:                  "unfulfilled",
-				Signature:               signature,
-				BlockNumber:             blockNumber,
-				RawOrder:                in["order"],
-				RawSignature:            in["signature"],
+				// Salt:                    order.Salt.Int64(),
+				Status:       "unfulfilled",
+				Signature:    signature,
+				BlockNumber:  blockNumber,
+				RawOrder:     in["order"],
+				RawSignature: in["signature"],
 			}
 			lotp.memoryDb.Add(limitOrder)
 		}
 		if m.Name == "executeMatchedOrders" && checkTxStatusSucess(backend, tx.Hash()) {
 			signatures := in["signatures"].([2][]byte)
 			fillAmount := uint(in["fillAmount"].(*big.Int).Int64())
+			// lotp.memoryDb.UpdatePositionForOrder(string(signature1), float64(fillAmount))
+			// lotp.memoryDb.UpdatePositionForOrder(string(signature2), float64(fillAmount))
+
 			lotp.memoryDb.UpdateFilledBaseAssetQuantity(fillAmount, signatures[0])
 			lotp.memoryDb.UpdateFilledBaseAssetQuantity(fillAmount, signatures[1])
 		}
+		if m.Name == "settleFunding" {
+			// funding payment was successful, so update the nnext funding time now
+			lotp.memoryDb.UpdateNextFundingTime()
+		}
+		if m.Name == "addMargin" {
+			// funding payment was successful, so update the nnext funding time now
+			// lotp.memoryDb.UpdateMargin()
+		}
 	}
 }
 
+func (lotp *limitOrderTxProcessor) ExecuteLiquidation(trader common.Address, matchedOrder LimitOrder) error {
+	nonce := lotp.txPool.Nonce(common.HexToAddress(userAddress1)) // admin address
+
+	data, err := lotp.orderBookABI.Pack("liquidateAndExecuteOrder", trader.String(), matchedOrder.RawOrder, matchedOrder.Signature)
```

Comment:
in contracts, we have modified this call such that now the `fillAmount` also needs to be passed in; which will be `min(order unfilled size, portion of liquidable position)`. This is will also help us track size so that we can continue to use that order for further matching - fdgod's latest code lets us track unfilled order quantity.